QuickFIXn/Logger/NonSessionFileLogger.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace QuickFix.Logger;
+
+/// <summary>
+/// Like the file logger, but only creates the files on first write
```

Comment:
so if I understand it, if FileLog were fixed to not create a file immediately, this class would not be needed